### PR TITLE
Dgan config and notebook

### PR DESCRIPTION
**Pull request by santhosh97**
DGAN Configuration File and Notebook

### CODE REVIEW

**Review by kboyd — APPROVED** [review]
I've added several commits resolving any comments I had.

So I'm approving, but should get at least a non-author approval too.

File: config_templates/gretel/synthetics/dgan.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  - timeseries_dgan:
+        data_source: "_"
+
+        # Replace with the column name containing the date or time
+        # values in your dataset.
+        time_column: "Date"
+        df_style: "long"
+        params:
+            sample_len: 1
+            max_sequence_len: 10
```
Comment (belson17) [inline config_templates/gretel/synthetics/dgan.yml:15]: If this (or others) variable is very likely to need to change for a user's dataset, then I think we should have a comment about it the same way we do for the time_column.

File: docs/notebooks/create_synthetic_data_from_dgan_api.ipynb [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+        "            discriminator_learning_rate: 1e-5\n",
+        "            attribute_discriminator_learning_rate: 1e-5\n",
+        "            batch_size: 100\n",
+        "            epochs: 5000\n",
+        "            apply_example_scaling: False\n",
+        "            attribute_loss_coef: 10\n",
+        "\"\"\"\n",
+        "\n",
+        "\n",
+        "# TODO: Change to use config template once template is available.\n",
```
Comment (belson17) [inline docs/notebooks/create_synthetic_data_from_dgan_api.ipynb:191]: Still need to update this TODO.

Comment (kboyd) [reply docs/notebooks/create_synthetic_data_from_dgan_api.ipynb:191]: We can change it with this PR, but then the notebook won't run until the config template is in the prod branch of this repo (this PR includes the config template but goes to the develop branch, so we can't test the notebook). So it's a bit awkward with the notebook and config in the same PR.

**Review by belson17 — APPROVED** [review]

File: docs/notebooks/create_synthetic_data_from_dgan_api.ipynb (hunk 2 above)
Comment (belson17) [reply docs/notebooks/create_synthetic_data_from_dgan_api.ipynb:191]: Got it, let's do another PR after this one then?

**Pull request merged**